activesupport/lib/active_support/cache.rb

```diff
@@ -802,11 +802,11 @@ class Entry # :nodoc:
 
       # Creates a new cache entry for the specified value. Options supported are
       # +:compress+, +:compress_threshold+, +:version+ and +:expires_in+.
-      def initialize(value, compress: true, compress_threshold: DEFAULT_COMPRESS_LIMIT, version: nil, expires_in: nil, **)
```

Comment:
From what I can tell this `**` was added so that extra keyword arguments would be ignored: https://github.com/rails/rails/pull/32539/files#diff-545287677174ddf9aff59851be5b659a050f2ff5db316c0a9a57dcd24b72dcd0R716, probably because `Entry` used to take an `options` hash.

I don't think it's an important compatibility breakage to not longer except unknown options, if anything it's more likely to catch some mistakes.

Reply:
Ok, so to answer myself, removing it breaks a bunch of things because the options of most methods are entirely forwarded to `Entry.new` in several places. So I've put it back.

This means that this PR basically adds a new feature: `Rails.cache.write("foo", expires_at: 2132131223)`, which I suppose is ok, but might need to be documented as well as handle `Time` instances.

Reply:
It seems like there might be some ambiguity around what happens if you pass both `expires_in` _and_ `expires_at`. You're prioritizing `expires_at` which makes sense I guess, but I wonder if both being passed should just be disallowed (`raise` or something).

Reply:
I don't think that's possible because you can set a default `expires_in` when you instantiate the `Store`, and it will be always forwarded to the `Entry`.

I guess we could handle that in the option merging code though.